### PR TITLE
global: addition of the ``cache`` module

### PR DESCRIPTION
**Pull request by drjova**
* Adds the ``cache`` module with a ``lock_cache`` proxy for
  easy access to the lock cache.

Signed-off-by: Harris Tzovanakis <me@drjova.com>

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/modules/cache/__init__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+"""INSPIRE cache."""
+
+from __future__ import absolute_import, division, print_function
+
+from .proxies import lock_cache
+
+__all__ = (
+    'lock_cache',
+)
```
Comment (jacquerie) [inline inspirehep/modules/cache/__init__.py]: I never understood why Invenio does this. This only matters when you do `from inspirehep.modules.cache import *`, which you never want to do, so why do they do it?

File: tests/integration/cache/test_lock_cache.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    lock_cache.set('Defenders', 'Jessica Jones, Luke Cage')
+    # expected value
+    expected = 'Jessica Jones, Luke Cage'
+    # get the value
+    result = lock_cache.get('Defenders')
+
+    assert expected == result
+
+
+def test_lock_cache_deleteion():
```
Comment (jacquerie) [inline tests/integration/cache/test_lock_cache.py]: `s/deleteion/deletion`

File: tests/integration/cache/test_lock_cache.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+"""Test lock cache."""
+
+from __future__ import absolute_import, division, print_function
+
+from inspirehep.modules.cache import lock_cache
+
+
+def test_lock_cache_creation(app):
+    """Test set key."""
+    # set the key
```
Comment (jacquerie) [inline tests/integration/cache/test_lock_cache.py]: Not a fan of this kind of comments, but not that big of a deal either.

File: tests/integration/cache/test_lock_cache.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Test lock cache."""
```
Comment (jacquerie) [inline tests/integration/cache/test_lock_cache.py]: Also not a fan of docstrings in tests (they tend to go stale really quickly!), but also not a big deal either.

**Comment by david-caro** [comment]
> some stylistical differences

I think we are due a meeting to agree on these

File: inspirehep/modules/cache/__init__.py (hunk 1 above)
Comment (drjova) [reply inspirehep/modules/cache/__init__.py]: just to make sure you're exporting what you need, but I agree it doesn't make sense in this case :)

**Comment by drjova** [comment]
updated and rebased, thanks!

File: tests/integration/cache/test_lock_cache.py (hunk 3 above)
Comment (drjova) [reply tests/integration/cache/test_lock_cache.py]: are just for readability, I don't have a strong opinion for them I will remove them :)

File: inspirehep/modules/cache/providers/lock.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+        Args:
+            key (str): a key name.
+
+        Returns:
+            str: the value of the given key.
+        """
+        return current_cache.get(self._prefix(key))
+
+    def set(self, key, value, timeout=300):
```
[the remark this answers is not in the document]
Comment (drjova) [reply inspirehep/modules/cache/providers/lock.py]: 👍 IMHO 2h is too much, we've discussed with @StellaCh and 30mins (of inactivity) should be enough.

[the remark this answers is not in the document]
Comment (david-caro) [reply inspirehep/modules/cache/providers/lock.py]: Make sure again, I got the feedback from the catalogers directly (annete, kirsten, florian, heath...)

**Comment by drjova** [comment]
ready to :shipit:

**Pull request merged**